cmd/clusterawsadm/cmd/resource/list/list.go

```diff
@@ -0,0 +1,143 @@
+			is not provided, then it will throw an error.
+		`),
+		Example: cmd.Examples(`
+		# List AWS resources created by CAPA in given region and clustername
+		clusterawsadm resource list --region=us-east-1 --cluster-name=test-cluster
+		`),
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			region, err = flags.GetRegion(cmd)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("Attempting to fetch resources created by CAPA for cluster:%s present in %s\n\n", clusterName, region)
+			return listAWSResource(&region, &clusterName, &outputPrinter)
+		},
+	}
+
+	newCmd.Flags().StringVarP(&region, "region", "r", "", "The AWS region where resources are created by CAPA")
+	newCmd.Flags().StringVarP(&clusterName, "cluster-name", "n", "", "The name of the cluster where AWS resources created by CAPA")
+	newCmd.Flags().StringVarP(&outputPrinter, "output", "o", "table", "The output format of the results. Possible values: table,json,yaml")
+	newCmd.MarkFlagRequired("cluster-name") //nolint: errcheck
+	return newCmd
+}
+
+func listAWSResource(region, clusterName, printerType *string) error {
+	cfg := aws.Config{}
+	if *region != "" {
+		cfg.Region = region
+	}
+
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+		Config:            cfg,
+	})
+	if err != nil {
+		return err
+	}
+
+	resourceClient := tagapi.New(sess)
+	input := &tagapi.GetResourcesInput{
+		TagFilters: []*tagapi.TagFilter{},
+	}
+
+	awsResourceTags := infrav1.Build(infrav1.BuildParams{
+		ClusterName: *clusterName,
+		Lifecycle:   infrav1.ResourceLifecycleOwned,
+	})
+
+	for tagKey, tagValue := range awsResourceTags {
+		tagFilter := &tagapi.TagFilter{}
+		tagFilter.SetKey(tagKey)
+		tagFilter.SetValues([]*string{&tagValue})
+		input.TagFilters = append(input.TagFilters, tagFilter)
+	}
+
+	output, err := resourceClient.GetResources(input)
+	if err != nil {
+		return err
+	}
+
+	if len(output.ResourceTagMappingList) == 0 {
+		fmt.Println("Could not find any AWS resource created by CAPA")
+		return nil
+	}
+
+	resourceList := awsResourceList{
+		ClusterName:  *clusterName,
+		AWSResources: []awsResource{},
+	}
+
+	for _, eachResource := range output.ResourceTagMappingList {
+		resourceARN, err := arn.Parse(*eachResource.ResourceARN)
+		if err != nil {
+			return err
+		}
+		eachAWSResource := awsResource{
+			Partition: resourceARN.Partition,
+			Service:   resourceARN.Service,
+			Region:    resourceARN.Region,
+			AccountID: resourceARN.AccountID,
+			Resource:  resourceARN.Resource,
+			ARN:       *eachResource.ResourceARN,
+		}
+		resourceList.AWSResources = append(resourceList.AWSResources, eachAWSResource)
+	}
+
+	outputPrinter, err := cmdout.New(*printerType, os.Stdout)
+	if err != nil {
+		return fmt.Errorf("failed creating output printer: %w", err)
+	}
+
+	fmt.Printf("Following resources are found:\n\n")
```

Comment:
Having output to stdout like this would break if you wanted json or yaml output, so switch to stderr output for this line